src/components/Divider.tsx

```diff
@@ -1,50 +1,79 @@
+const DIVIDER_MIN_WIDTH = 85
 
 /** A custom horizontal rule. */
 const Divider = ({ path }: { path: Path }) => {
-  const dividerSetWidth = React.createRef<HTMLInputElement>()
   const dispatch = useDispatch()
+  const dividerRef = createRef<HTMLDivElement>()
+  const thoughtIndex = useSelector((state: State) => state.thoughts.thoughtIndex)
+  const [width, setWidth] = useState(DIVIDER_MIN_WIDTH)
 
   /** Sets the cursor to the divider. */
   const setCursorToDivider = (e: React.MouseEvent | React.TouchEvent) => {
     e.stopPropagation()
     dispatch(setCursor({ path }))
   }
 
-  /** Get the max width of nearby for divider list child elements, add 30 px and set this width for divider. */
+  /** Get the max width of nearby elements, add DIVIDER_PLUS_PX and set this width for divider. */
   const setStyle = () => {
-    if (dividerSetWidth.current) {
-      const parentUl = dividerSetWidth.current.closest('ul')
-      const children = parentUl ? (Array.from(parentUl.childNodes) as HTMLElement[]) : []
-      const widths = children.map((child: HTMLElement) => {
-        if (child.classList.contains('child-divider')) return DIVIDER_PLUS_PX
-        const subs = child.getElementsByClassName('subthought') as HTMLCollectionOf<HTMLElement>
-        return subs.length ? subs[0].offsetWidth + DIVIDER_PLUS_PX : DIVIDER_PLUS_PX
-      })
-      const maxWidth = Math.max(...widths)
-      dividerSetWidth.current.style.width = `${maxWidth > DIVIDER_MIN_WIDTH ? maxWidth : DIVIDER_MIN_WIDTH}px`
+    if (dividerRef.current) {
+      const parentPath = parentOf(path)
+
+      const treeNode = dividerRef.current.closest('div.tree-node') as HTMLElement
+      if (!treeNode) throw new Error('Divider tree node not found')
+
+      const thoughtPath = treeNode.dataset.path
+      if (!thoughtPath || thoughtPath === '') throw new Error('Divider path not found on tree node')
+
+      let elements = document.querySelectorAll(`.tree-node[data-path^="${hashPath(parentPath)}"]:not(.thought-divider)`)
+
+      if (elements.length === 1) {
+        /** If this divider is an only child, search further up the tree. */
+        const grandparentPath = parentOf(parentPath)
+        if (!grandparentPath) throw new Error('Divider grandparent not found')
+
+        elements = document.querySelectorAll(
+          `.tree-node[data-path^="${hashPath(grandparentPath)}"]:not(.thought-divider)`,
+        )
+      } else if (!elements.length) {
+        /** This is a top-level divider, use its siblings instead. */
+        elements = document.querySelectorAll('.tree-node.top-level:not(.thought-divider)')
+      }
+
+      /** Find and set the max width of our selected elements. */
+      const maxWidth = Math.max(
+        ...Array.from(elements).map(element => (element.querySelector('.thought') as HTMLElement).offsetWidth),
+      )
+
+      setWidth(maxWidth + DIVIDER_PLUS_PX)
     }
   }
 
-  useEffect(setStyle)
+  useEffect(setStyle, [setStyle, thoughtIndex])
```

Comment:
`setStyle` is not a stable ref, so this will cause `useEffect` to trigger on every render. This is probably masking a missing dependency.

The `thoughtIndex` dependency is good because at least we're getting the additional render that is needed when the thoughts load.

Reply:
Oops, I'm not even sure why I did that. I pushed the fix, although it didn't affect the number of times setStyle got called, but everything seems kosher now.

Reply:
Confusingly, the ref container is not actually a stable reference, only the object inside:

```ts
dividerRef // unstable
dividerRef.current // stable
```

So the number of times `setStyle` is called hasn't changed because it's still getting called on every render.

Reply:
The only issue with that is I get this complaint:

![dividerRef_dependency](https://github.com/user-attachments/assets/17e973b9-0ef9-434b-83bf-ebcbeaf917fb)

Reply:
If `dividerRef` is unstable, and `dividerRef.current` is an invalid dependency, then what do we do? I must be missing something. 

Maybe you can find the answer online. We can't be the first ones to run into this.

Reply:
Right. I'll let you know what I come up with.

Reply:
56171ea - After some trial and error, I think the cleanest way I can come up with is to just bring back the eslint-disable for exhaustive-deps, and at that point we don't really need to rely on the `dividerRef` at all since it was just added to satisfy the eslint warning. Not having the ref as a dependency calls `setStyle` the same number of times as using `dividerRef.current` so we still have stable dependencies and preserve the correct behavior on page load with the divider selected. Does this seem effective enough to you, or maybe I'm missing something else?

Reply:
Sure, we can suppress the linter for now. There is a chance that the DOM may update and `setStyle` will not be triggered, but I don't have a test case for it.

FYI It looks like the recommended solution is to switch from an object ref to a callback ref, but my experiments didn't show that it made much difference. It will be good to keep these in mind if we run into this issue in a different context though:

- https://stackoverflow.com/questions/60476155/is-it-safe-to-use-ref-current-as-useeffects-dependency-when-ref-points-to-a-dom
- https://medium.com/@teh_builder/ref-objects-inside-useeffect-hooks-eb7c15198780
- https://legacy.reactjs.org/docs/hooks-faq.html?source=post_page-----eb7c15198780--------------------------------#how-can-i-measure-a-dom-node

Reply:
I added a comment in 7e7e4b5c04 so that future developers can understand why we suppressed the linter.